src/FFTNormalOp.jl

```diff
@@ -73,21 +78,25 @@ struct _FFTNormalOp{S,ΛType,T,N,E,F,G}
     cmaps::G
 end
 
-function calculateKernelBasis(img_shape, trj, U)
+function calculate_kernel_cartesian(img_shape, trj, U; mask=trues(size(trj)[2:end]))
     Ncoeff = size(U, 2)
     Λ = zeros(eltype(U), Ncoeff, Ncoeff, img_shape...)
 
     Threads.@threads for ic ∈ CartesianIndices((Ncoeff, Ncoeff))
-        for it ∈ axes(U, 1), ix ∈ axes(trj[it], 2), irep ∈ axes(U, 3)
-            k_idx = ntuple(j -> mod1(Int(trj[it][j, ix]) - img_shape[j] ÷ 2, img_shape[j]), length(img_shape)) # incorporates ifftshift
-            k_idx = CartesianIndex(k_idx)
-            Λ[ic[1], ic[2], k_idx] += conj(U[it, ic[1], irep]) * U[it, ic[2], irep]
+        for it ∈ axes(U, 1), ix ∈ axes(trj, 2)
+            if mask[ix, it]
+                k_idx = ntuple(j -> mod1(Int(trj[j, ix, it]) - img_shape[j] ÷ 2, img_shape[j]), length(img_shape)) # incorporates ifftshift
+                k_idx = CartesianIndex(k_idx)
+                for irep ∈ axes(U, 3)
```

Comment:
What does `irep` stand for here? Is there a better variable name? Similar for `ix`.